Modify library structure to support web components

flemcito opened the pull request:
This PR closes #1315 

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

A major restructuring of the repo. 
- Introducing the @kirbydesign/core package. 

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [x] Enhancement (to existing content)
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [x] CI related changes
- [x] Documentation content changes
- [x] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## Which issue documents the current behaviour?

<!-- Please link to a relevant issue that documents the current behaviour . -->

Issue Number: #[replace with issue number]

## What is the new behavior?

<!-- Please describe the new behaviour after your pull-request is comitted -->

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

flemcito committed:
stencil scaffold

flemcito committed:
mocks

flemcito committed:
Merge branch 'master' into enhancement/web-components

flemcito committed:
Update mock-components.ts

flemcito committed:
First spike

flemcito committed:
with gitignore

flemcito committed:
Merge branch 'master' into enhancement/web-components

flemcito committed:
deleted old badge

flemcito committed:
Update mock-components.ts

flemcito committed:
removed ref to old badge

flemcito committed:
Update mock-components.ts

flemcito committed:
trying to get test to work

flemcito committed:
Update mock-components.ts

flemcito committed:
Merge branch 'master' into enhancement/web-components

flemcito committed:
Update mock-components.ts

flemcito committed:
built new core

flemcito committed:
Update mock-components.ts

flemcito committed:
cleanup config

flemcito committed:
Update mock-components.ts

flemcito committed:
test experiments

flemcito committed:
Update mock-components.ts

flemcito committed:
exclude specs in tsconfig

flemcito committed:
Update mock-components.ts

flemcito committed:
link core

flemcito committed:
Update mock-components.ts

flemcito committed:
Merge branch 'master' into enhancement/web-components

flemcito committed:
Update mock-components.ts

flemcito committed:
Fixed e-2-e tests

flemcito committed:
bump node version in ci

flemcito committed:
node version on ci

flemcito committed:
Update mock-components.ts

flemcito committed:
try to fix ci build

flemcito committed:
Update mock-components.ts

flemcito committed:
removed autogenerated files

flemcito committed:
Update mock-components.ts

flemcito committed:
more ci

flemcito committed:
updated prettier ignore

flemcito committed:
Merge branch 'master' into enhancement/web-components

flemcito committed:
fun with ts config WIP

flemcito committed:
Update mock-components.ts

flemcito committed:
removed tsconfig tightening (seperate issue)

flemcito committed:
Update mock-components.ts

flemcito committed:
Moved styling to core

flemcito committed:
Update mock-components.ts

flemcito committed:
Merge branch 'master' into enhancement/web-components

flemcito committed:
Update mock-components.ts

flemcito committed:
Update package-lock.json

flemcito committed:
Update mock-components.ts

flemcito committed:
try adding symlink

flemcito committed:
Update mock-components.ts

flemcito committed:
yet another attempt

flemcito committed:
Update mock-components.ts

flemcito committed:
Merge branch 'master' into enhancement/web-components

flemcito committed:
readded stencil deps

flemcito committed:
Merge branch 'master' into enhancement/web-components

flemcito committed:
adjust imports

flemcito committed:
brought styling up to date

RasmusKjeldgaard committed:
🔥 Remove toolversions file

RasmusKjeldgaard committed:
✅  Refactor test to support querying shadow root

RasmusKjeldgaard committed:
🎨  Move core .gitignore to existing .gitignore

RasmusKjeldgaard committed:
🔧 Clean up prettierignore paths

RasmusKjeldgaard committed:
🚨 Add unmodified autogenerated proxies file

RasmusKjeldgaard committed:
🔧  Remove compiler settings to make stencil build

RasmusKjeldgaard committed:
⬆️  Update jest and puppeteer types

RasmusKjeldgaard committed:
Merge branch 'master' into enhancement/web-components

RasmusKjeldgaard committed:
🚨 Make badge import in module specific

RasmusKjeldgaard committed:
➕ Add babel dev dependency to fix cookbook compile

RasmusKjeldgaard committed:
🔧 Add core lib tests to pre push hook

RasmusKjeldgaard committed:
🔥 Remove editorconfig

RasmusKjeldgaard committed:
🔥 Remove autogenerated e2e test

RasmusKjeldgaard committed:
🔨 Use watchall on default core test script

RasmusKjeldgaard committed:
🔧 Remove unused json docs output target

RasmusKjeldgaard committed:
🔨 Add single run build script for pre-push hook

RasmusKjeldgaard committed:
⏪ Revert unrelated changes to form field showcase

RasmusKjeldgaard commented:
TODO:
[] Create separate branch for autogenerating mock based on this branch, and merge to this
[] **!IMPORTANT!** Before merge, make sure to copy over the content of the global styles that have been moved, so they match the current state of master

RasmusKjeldgaard committed:
🔥 Move global styles to core

RasmusKjeldgaard committed:
💄 Reference new util file for all cookbook examples

RasmusKjeldgaard committed:
💡 Add TODO comment to old utils file

RasmusKjeldgaard committed:
✅  Fix badge name in spec describe

RasmusKjeldgaard committed:
🚚  Move typography spec and remove base scss folder

RasmusKjeldgaard commented on libs/designsystem/src/lib/helpers/color-helper.scss:
```diff
@@ -1,4 +1,4 @@
 @use 'sass-to-ts';
-@use '../scss/themes/colors';
+@use '../../../../core/src/scss/themes/colors';
```
use `~@kirbydesign/core` alias

RasmusKjeldgaard commented on libs/designsystem/src/lib/helpers/design-token-helper.scss:
```diff
@@ -1,5 +1,5 @@
 @use 'sass-to-ts';
-@use '../scss/base/variables';
+@use '../../../../core/src/scss/base/variables';
```
use `~@kirbydesign/core` alias

RasmusKjeldgaard commented on libs/designsystem/src/lib/scss/_utils.scss:
```diff
@@ -1,8 +1,2 @@
-@import './base/include-media';
-@import './themes/colors';
+// TODO: Remove this file when all components are moved to core
+@import '../../../../core/src/scss/utils';
```
use ~@kirbydesign/core alias

RasmusKjeldgaard commented on libs/core/stencil.config.ts:
```diff
@@ -0,0 +1,20 @@
+    {
+      type: 'dist',
+      esmLoaderPath: '../loader',
+      // copy: [{ src: './scss', dest: '../scss' }],
```
Remove comment

RasmusKjeldgaard commented on libs/designsystem/tslint.json:
```diff
@@ -3,5 +3,8 @@
     "component-selector": [true, "element", "kirby", "kebab-case"]
+  },
+  "linterOptions": {
+    "exclude": ["src/lib/components/angular-component-lib/**.*"]
```
remember to change this with new /stencil/ structure

RasmusKjeldgaard commented on package.json:
```diff
@@ -160,7 +175,7 @@
     "hooks": {
       "pre-commit": "npm run sync-node-version && npm run generate-mocks && lint-staged",
-      "pre-push": "npm run test:headless-browser:single",
+      "pre-push": "npm run test:core:single && npm run test:headless-browser:single",
```
extract these two run steps into `test:libs` or similar

RasmusKjeldgaard commented on tsconfig.base.json:
```diff
@@ -8,9 +8,10 @@
     "experimentalDecorators": true,
-    "target": "es2015",
+    "allowUnreachableCode": false,
+    "target": "es2017",
```
why is this needed? Check up on consequences.

RasmusKjeldgaard replied:
The way I see it, this change is sane enough. 

This has no consequences for the angular library, as the tsconfig for that sets the target to `es2017`. 
So instead, we could do as suggested here and set the base config to `es2017`, and remove it from the angular specific config. 

Furthermore, the cookbook has a browserslistrc that angular uses to determine the needed JS output. Running `npx browserslist`, I see the lowest supported versions as:

```
chrome 90
edge 89
firefox 78
ios_saf 13.0-13.1
safari 13
```
Which is all well within [support for es2017](https://caniuse.com/?search=es2017).

RasmusKjeldgaard replied:
We should consider to bump the target of tsconfig.lib.json (published angular lib) to es2017 since we are already 'promising' this and more by setting the 'lib' to es2018. To my knowledge we are not supplying polyfills to close that gap.

RasmusKjeldgaard commented on tsconfig.base.json:
```diff
@@ -8,9 +8,10 @@
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
-    "target": "es2015",
+    "allowUnreachableCode": false,
```
remove

RasmusKjeldgaard commented on tsconfig.base.json:
```diff
@@ -8,9 +8,10 @@
+    "target": "es2017",
     "typeRoots": ["node_modules/@types"],
-    "lib": ["es2017", "dom", "es6", "es2015.iterable"],
+    "lib": ["es2017", "dom"],
```
What is happening here?

RasmusKjeldgaard replied:
The target (previously `es2015`) is supplemented by the `lib` option which is used to indicate what extra JS API’s/features the compiled project uses on top of `es2015`. So if we supply polyfills for some or all `es2017`-specific features we can indicate that in the `lib` config. 

But I can only think of use cases where we want to compile to a lower target like `es2015` to support older browsers, but then supply polyfills for some features, so the runtime can be es2017. And to silence the compiler we then specify what functionality we ensure ourselves to deliver through the `lib` config. 

Since our previous target was `es2015` (which is also es6), there should never have been a need for these, as we already have a target that supports these features. 

In conclusion I cant see anything wrong with removing this. It is overwritten for our published package, so it is only relevant for the compiled cookbook code, and the core library.

https://www.typescriptlang.org/tsconfig#lib
https://stackoverflow.com/a/42097465
https://www.meziantou.net/which-version-of-ecmascript-should-i-use-in-the-typescript-configuration.htm

RasmusKjeldgaard replied:
We should consider to bump the target of [tsconfig.lib.json] (https://github.com/kirbydesign/designsystem/blob/master/libs/designsystem/tsconfig.lib.json) (published angular lib) to `es2017` since we are already 'promising' this and more by setting the 'lib' to `es2018`. To my knowledge we are not supplying polyfills to close that gap.

RasmusKjeldgaard committed:
🚚  Add '.component' to filenames to indicate their scope

RasmusKjeldgaard committed:
💄 Restore scss to make git recognize moved file

RasmusKjeldgaard committed:
💄 Reintroduce badge web comp styling

RasmusKjeldgaard committed:
Generate Angular mocks for new web-components (#1677)

* 🔨  Handle barrel file type assertions during mock generation

* 🔨 Generate class decoration from asserted type

* 🔨 Simplify if-else with ternary operator

* 🔨 Generate custom-elements json doc

* 🔧 Rename generated custom elements file

* 🔨 Use custom elements json for mock generation

* ✨ Add MockBadgeComponent

* 🔨 Build core before transpiling tools so custom elements files are available

* 🔨 Generate mocks on changes to core lib

* 🎨  Simplify custom elements component traversing code

* 🔨 Change inputpaths to object and describe in comments

* 🔨 Rename proxies file

* ♻️ Split typesInFile functionality to own function

* ♻️ Reduce responsibility of getExportedTypes

* ♻️ Replace use of 'asserted' with 'alias'

* ♻️ Simplify use of aliasmap when traversing components

* 🔥 Remove mock files after component file rename

* ♻️ Make aliasmap internal to mapping func and return as tuple

* ♻️  Rename mapAndRemoveAliases -> separateTypesFromAliases

* ♻️ Pass aliasMap for providers to reduce optional chaining

* ♻️  Rename type and alias related functions

Co-authored-by: RasmusKjeldgaard <rkk@bankdata.dk>

RasmusKjeldgaard committed:
✅  Import Ionic module through TestHelper in badge spec

RasmusKjeldgaard committed:
💄 Reference core styles via core package

RasmusKjeldgaard committed:
♻️ Move custom elements init to function for reuse

RasmusKjeldgaard committed:
🚚 Move design token, color and string helpers to core

The rest of the helpers are still placed under the angular project.
These are to be moved when components using each helper is moved to core

RasmusKjeldgaard committed:
💄 Reference core styles from angular component scss

RasmusKjeldgaard committed:
💄 Reference core styles from cookbook

RasmusKjeldgaard committed:
🎨  Create barrel files to support new helpers in core

RasmusKjeldgaard committed:
📦  Update tests to use core helpers

RasmusKjeldgaard committed:
📦  Update components to use core helpers

RasmusKjeldgaard committed:
Merge branch 'master' into enhancement/web-components

RasmusKjeldgaard committed:
➕ Update package lock to match package.json

RasmusKjeldgaard committed:
💄 Change cookbook utils imports to core module

RasmusKjeldgaard committed:
💚  Run build step after extracting scss variables

RasmusKjeldgaard committed:
💚 Transpile mock script after custom elm docs gen

When transpiling the mock script, the build command for stencil
components needs to be run first, so custom-elements.json is available.

The rest of the transpilation needs to happen before build, so we can
extract scss variables (.styles.ts files) and use them in components.

RasmusKjeldgaard committed:
💚 Fix build by explicitly referencing core helpers

RasmusKjeldgaard committed:
✨ Use helper for animations to keep .styles.ts internal

Before, we referenced the .styles.ts directly but now that file is moved
out of the source files of the angular project. Therefore we need to
provide the styles through the DesignTokenHelper.

RasmusKjeldgaard committed:
📦  Use core helpers in new chart component

RasmusKjeldgaard committed:
🔧Add combined test step for testing both packages

RasmusKjeldgaard commented on libs/designsystem/src/lib/components/angular-component-lib/utils.ts:
```diff
@@ -0,0 +1,43 @@
+import { fromEvent } from 'rxjs';
+
```
Not related to specific line: 
Seems like the path for this (angular-component-lib) is not configurable. It is autogenerated by stencil, based on where the proxies.ts file is outputted.

RasmusKjeldgaard committed:
🔧 Remove unreachable code flag as it is deault

RasmusKjeldgaard committed:
Revert "🔧Add combined test step for testing both packages"

This reverts commit 01fe8124352194f5f30bfbe3f9f155564ea63f35.

RasmusKjeldgaard committed:
🔧 Add combined test step for testing both packages

RasmusKjeldgaard commented on package.json:
```diff
@@ -24,7 +24,7 @@
     "ng": "nx",
     "nx": "nx",
-    "start": "ng serve",
+    "start": "ng serve && npm run build:core:watch",
```
This is not going to work - ng serve does not terminate, so core is never built. 

Consider concurrently, npm-run-all or similar to run the processes besides each other.

RasmusKjeldgaard committed:
➖ Remove puppeteer dependency

RasmusKjeldgaard commented:
TODO:
- [ ] Find an alternative to themeColor directive, as it is angular specific
- [x] Create separate branch for autogenerating mock based on this branch, and merge to this
- [ ] Create separate branch for publishing (#1630) and merge to this 
- [ ] Create both npm packages locally and install in DRB as test
- [ ] **!IMPORTANT!** Before merge, make sure to verify that moved helpers and style files have any new changes from master.

RasmusKjeldgaard committed:
♻️ Add themeColor prop to badge

A prop is now used directly within the badge component, instead
of being added via a directive, to support setting themeColor outside
of Angular projects.

Badge does not use the colorBrightness logic added via the directive
for anything and therefore this change should be safe.

RasmusKjeldgaard committed:
🔨 Add '/' to generate-mock core path for consistency

RasmusKjeldgaard committed:
Merge branch 'master' into enhancement/web-components

RasmusKjeldgaard commented on libs/core/src/components/badge/badge.component.tsx:
```diff
@@ -0,0 +1,21 @@
+import { Component, ComponentInterface, h, Host, Prop } from '@stencil/core';
+
+import { ColorHelper, ThemeColor } from '../../helpers';
```
Remove unused ```ColorHelper```- and set up eslint?

RasmusKjeldgaard committed:
Merge branch 'master' into enhancement/web-components

RasmusKjeldgaard committed:
💄 Reference core util scss file for layout recipe

RasmusKjeldgaard committed:
🔥 Remove unused ColorHelper reference

RasmusKjeldgaard committed:
➕ Use npm-run-all to build and serve in parallel

RasmusKjeldgaard committed:
🎨 Add whitespace for readability

RasmusKjeldgaard committed:
Merge branch 'master' into enhancement/web-components

RasmusKjeldgaard committed:
Reference BadgeComponent directly in lib index.ts to mitigate UMD compilation bug (#1773)

* 🚧  Temporarily fix UMD compilation bug

By not asserting the type of KirbyBadge to be BadgeComponent, the
UMD-compilation bug in DRB is mitigated. For now, the badge component
is exported as KirbyBadge, but this will most likely cause problems
with mock generation, as consumers expect the component to be named
MockBadgeComponent, not MockKirbyBadge

* Revert "🚧  Temporarily fix UMD compilation bug"

This reverts commit 5010a3038e4fac73bd92802e197c785d6413495f.

* 🐛  Move BadgeComponent export one level up

This change is needed to avoid an UMD compilation bug in consumer code.
Previously, when compiling the UMD bundle provided by Kirby, consumers
would experience a 'RangeError: Maximum call stack size exceeded'.

We are still investigating the actual cause of this, but this simple change
mitigates the issue.

* 🚨 Change spec import to match new export location

* 🎨 Add comment to explain change

RasmusKjeldgaard committed:
Added publish and version bump workflow for core package (#1722)

* 🔨  Copy core scss to angular package on publish

* 🔧 Copy scss folder to root of core package

In order to be able to use scss files such as global styles in a project
via '@kirbydesign/core/scss' the folder is copied to the root of
the package. This can be handled by the stencil compiler, so as a
trade off, it will be done every time the project is built.
Therefore, the folder has also been added to .gitignore.

* 🔧 Allow package to be published, and edit description

* 🚧  Add Github Actions workflow for core publish

* 🚧  Fix workflow error

* 🔥 Remove temporary test script

* 👷 Modify publish script to handle both packages

* 👷 Add back node setup and caching

* 👷 Change name of publish step

* ⬆️  Bump core version dependency in designsystem

* 🔨 Build core before creating local npm package

* ♻️ Refactor publish to match existing implementation

* 🎨 Improve naming and formatting of publish workflow

MadsBuchmann committed:
Merge branch 'master' into enhancement/web-components

MadsBuchmann committed:
:sparkles: Add zLayer designTokenHelper function

MadsBuchmann committed:
:recycle: Update scss utils path to new core structure

RasmusKjeldgaard commented:
Consider updating to [Stencil v.2.9.0](https://github.com/ionic-team/stencil/releases/tag/v2.9.0) before merging.

MadsBuchmann committed:
Add small badge component (#1812)

* :sparkles: Introduce 'size' input property

* :recycle: Redo 'badgeContent' getter

* :lipstick: Add styling for badge with size = sm

* :recycle: Move and rename getClassList & getClassMap

* :white_check_mark: Update test to include 'md' class

* :label: Create 'BadgeSize' type

* :white_check_mark: Add tests for small badge

* :recycle: Move 'BadgeSize' to separate file

* :memo: Split up badge examples

* :memo: Add text to the badge documentation

* :lipstick: Add cards around badge examples

* :lipstick: Add examples 'h2' heading

* Revert "Merge branch 'master' into enhancement/498-small-badge"

This reverts commit e38106ee56d5364a5a1dd50992d762d8d7f76733, reversing
changes made to 479015821aa958746361291220593fa9d3ec3f22.

* Revert "Revert "Merge branch 'master' into enhancement/498-small-badge""

This reverts commit 414d0a4d30651f041f9fc207f20eb0ecfaddc8f4.

* :white_check_mark: Fill in missing 'should have md class' test

* :memo: Add css custom properties table

* :pencil2: Change "inlines" to "inline"

* :memo: Remove 'but can be used to indicate status' part

* :memo: Have 'number' paragraph sound more like guidelines

* :memo: Add attachment example

* :pencil2: Change kirby-badge-zindex to kirby-badge-z-index

* :memo: Make proper spacing in popout & add elevation

* :bug: Fix elevation not displaying

* :memo: Add section regarding elevation on white badges

* :fire: Remove accidentally added double import of utils

* :lipstick: Have grey background extend all the way to bottom

* :bug: Fix wrong link & add icon

* :lipstick: Use 'kirby-external-icon' & move link to own line

RasmusKjeldgaard committed:
Add icon option to badge in segmented control item (#1774)

* ✨ Add slotted badge to segmented control

* Revert "✨ Add slotted badge to segmented control"

This reverts commit fb91c8e276468a5f7c3a8c8ccaf60bc70ac2d632.

* ✨ Add badge with icon to segment control

* ♻️  Make use of exposed badge custom properties

* 📝  Document icon badge in segmented control

* 📝  Update badge description in documentation

* 🐛 Make sure both icon and content does not render

Icon now takes precedence over content if both are present,
instead of rendering both elements.

* 💄 Revert unintentional change to badge elevation

* ✅  Add badge tests to segmented control

* 📝  Move SegmentedControl badge doc to own section

* 💄 Added proper external link styling to gh link

* 💄 Make link standalone to fit link guidelines

* ✅  Change wording of test

RasmusKjeldgaard committed:
💡 More concise comment for publish workflow

RasmusKjeldgaard committed:
💡 Fix gitignore designsystem file comment

RasmusKjeldgaard committed:
🔧 Clean up gitignore paths

RasmusKjeldgaard committed:
🎨 Update prettier config and run prettier

RasmusKjeldgaard committed:
⬆️  Bump version to match actual version on npm

RasmusKjeldgaard committed:
💄 Set default z-index for badge to auto

RasmusKjeldgaard committed:
💄 Use size token for badge size and border radius

RasmusKjeldgaard committed:
🔥 Remove test file - was just for experimenting

RasmusKjeldgaard committed:
Merge branch 'master' into enhancement/web-components

RasmusKjeldgaard committed:
🚚  Move print styles to core

RasmusKjeldgaard committed:
🚚  Rename appInitializer and dont export function

RasmusKjeldgaard committed:
🔧 Remove umdModuleID from ng-package

It is redundant, as ng-packagr will infer this on its own

RasmusKjeldgaard committed:
💄 Create z-index variable for badge in avatar

RasmusKjeldgaard committed:
💄 Use segmentBadge correctly so it evaluates

RasmusKjeldgaard committed:
🚨 reference customElementsInitializer correctly

RasmusKjeldgaard committed:
🎨 Add imports to same line

Co-authored-by: Mads Buchmann Frederiksen <mbf@prochimp.dk>

RasmusKjeldgaard committed:
🔥 Remove inlineSourceMap

RasmusKjeldgaard committed:
🔧 Update package.json description

RasmusKjeldgaard committed:
♻️ Rename libDir reference to angularLibDir

RasmusKjeldgaard committed:
🎨 Make imports one line

Co-authored-by: Mads Buchmann Frederiksen <mbf@prochimp.dk>

RasmusKjeldgaard committed:
💚 Add back export of appInitialize

We use it as a provider in KirbyModule and therefore it needs
to be exported

RasmusKjeldgaard committed:
🎨  Add whitespace to imports (tslint)

MadsBuchmann commented on apps/cookbook/src/app/examples/icon-example/icon-example.component.scss:
```diff
@@ -1,4 +1,4 @@
-@import './libs/designsystem/src/lib/scss/utils';
+@import '~@kirbydesign/core/src/scss/utils';
```
That is true. Looking at my suggestion again I think we should perhaps just ignore it? I have trouble seeing exactly what value it would add. It would also make it even more rigid to make changes to configuration/packaging, which I do not think is desirable.

MadsBuchmann commented on package.json:
```diff
@@ -24,16 +24,17 @@
     "ng": "nx",
     "nx": "nx",
-    "start": "ng serve",
+    "start": "npm-run-all --parallel \"ng serve\" \"build:core:watch\"",
```
Wooohooo!

MadsBuchmann commented on .github/workflows/publish.yml:
```diff
@@ -5,18 +5,19 @@ on:
-    if: github.event.pull_request.merged
     steps:
-      - name: Checkout master branch
+      - name: Checkout repo
```
Good point! 

We already have a issue on it here actually now that I turn on my brain: https://github.com/kirbydesign/designsystem/issues/1518 

I would LOVE to carry that one out.

MadsBuchmann commented on .gitignore:
```diff
@@ -9,6 +9,14 @@
+/libs/core/loader
+/libs/core/dist/
+/libs/core/scss/
+/libs/core/custom-elements.d.ts
+/libs/core/custom-elements.json
+/libs/core/src/components.d.ts
```
cool 😄

MadsBuchmann commented on .prettierignore:
```diff
@@ -1,2 +1,11 @@
+/libs/core/component-docs.d.ts
+/libs/core/component-docs.json
+/libs/core/src/components.d.ts
+readme.md
```
Ah cool :D!

MadsBuchmann commented on libs/core/package.json:
```diff
@@ -0,0 +1,15 @@
+  "module": "dist/index.js",
+  "main": "dist/index.cjs.js",
+  "types": "dist/types/index.d.ts",
+  "collection": "dist/collection/collection-manifest.json",
```
cool 🦖

MadsBuchmann commented on libs/designsystem/testing-base/src/lib/directives/mock.theme-color.directive.ts:
```diff
@@ -5,7 +5,6 @@ import { ThemeColor, ThemeColorDirective } from '@kirbydesign/designsystem';
 @Directive({
   // tslint:disable-next-line:directive-selector
   selector: `kirby-card[themeColor],
-             kirby-badge[themeColor],
```
Ah I misunderstood the selector in the first pass. Thank you for the clarification.

MadsBuchmann commented on libs/designsystem/src/lib/app-initialize.ts:
```diff
@@ -0,0 +1,19 @@
+
+import { defineCustomElements } from '@kirbydesign/core/loader';
+
+export const appInitialize = (doc: Document) => {
```
Cool 😄

MadsBuchmann commented on libs/designsystem/src/lib/components/avatar/avatar.component.scss:
```diff
@@ -21,7 +21,7 @@ $badge-diameter: $avatar-badge-size;
   --kirby-badge-right: #{get-badge-position($diameter-small)};
   --kirby-badge-top: #{get-badge-position($diameter-small)};
-  --kirby-badge-zindex: 2;
+  --kirby-badge-z-index: 2;
```
Sounds good. The z-index maps seems to be pretty... case specific?... 

So adding 'avatarBadge' does not seem to be poor form to me.

MadsBuchmann commented on libs/designsystem/tsconfig.lib.json:
```diff
@@ -4,6 +4,7 @@
     "outDir": "../../dist/out-tsc",
     "target": "es2015",
     "declaration": true,
+    "inlineSourceMap": true,
```
How did it go 😄?

RasmusKjeldgaard replied:
We have a silent compiler 🤐

MadsBuchmann commented on libs/designsystem/src/lib/animation/kirby-animation.ts:
```diff
@@ -1,12 +1,12 @@
-    LONG = parseInt(styles.transitionDurations.long),
-    EXTRA_LONG = parseInt(styles.transitionDurations.extraLong),
+    QUICK = parseInt(DesignTokenHelper.transitionDuration('quick')),
+    SHORT = parseInt(DesignTokenHelper.transitionDuration('short')),
+    LONG = parseInt(DesignTokenHelper.transitionDuration('long')),
+    EXTRA_LONG = parseInt(DesignTokenHelper.transitionDuration('extraLong')),
   }
-  export const Easing = styles.transitionEasings;
+  export const Easing = DesignTokenHelper.transitionEasings();
```
As is is just totally fine!

MadsBuchmann reviewed (dismissed):
It's an honor to finally be able to write these four letters with accompanying gif: 

LGTM!
![Happy Coder](https://media.giphy.com/media/l2JhC36VbivOHDAek/giphy.gif)

RasmusKjeldgaard committed:
:bookmark: Bumping version to 0.0.8 (core)

RasmusKjeldgaard committed:
Merge branch 'master' into enhancement/web-components
